flake.nix

```diff
@@ -103,7 +96,6 @@
               fi
             }
             PROMPT_COMMAND=prompt
-            ${pre-commit-check.shellHook}
```

Comment:
What about [this](https://github.com/IntersectMBO/cardano-ledger/pull/4234/files#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0L33): 
```
        pre-commit-check = inputs.pre-commit-hooks.lib.${system}.run {
          src = ./.;
          hooks = {
            fourmolu.enable = true;
          };
        };
```

Can we get rid of it as well?

Reply:
That one is still needed. Strictly speaking however, it is not required at that scope. The reason why it's there is because in #4175 I also added a check (see line ~244) that can be run and added the possibility to run `fourmolu` as a hydra job if we wish to. I can remove these if you want to and move the above `pre-commit-check` binding to `mkDevShells`.

Reply:
I added the changes in 009a5fe. Let me know if that is what you would prefer.

Reply:
Yes! I think it's better to limit the scope like you did, thank you!